src/recordlinker/schemas/pii.py

```diff
@@ -234,36 +234,31 @@ class Telecom(StrippedBaseModel):
     system: typing.Optional[str] = None
     use: typing.Optional[str] = None
 
-    def phone(self) -> str | None:
-        """
-        Return the phone number from the telecom record.
-        """
-        if self.system != "phone":
-            return None
-        # normalize the number to include just the 10 digits
-        return re.sub(r"\D", "", self.value).strip()[:10]
-
-    def email(self) -> str | None:
-        """
-        Return the email address from the telecom record.
-        """
-        if self.system != "email":
-            return None
-        return self.value.lower().strip()
-
-    @classmethod
-    @functools.lru_cache()
-    def get_system_handlers(cls) -> dict[str, str]:
-        """
-        Return a dictionary of system handlers for the Telecom class where the keys
-        are the system values and the values are the method names to call.
-
-        """
-        return {
-            name: name
-            for name, method in cls.__dict__.items()
-            if callable(method) and not name.startswith("_")
-        }
+    @pydantic.model_validator(mode="before")
+    def validate_and_normalize_telecom(cls, values):
+        """
+        Validate and normalize the telecom record.
+        """
+        # If telecom.system = "email", set telecom.value to lowercase
+        if values.get("system") == "email":
+            values["value"] = values["value"].strip().lower()
+        # If telecom.system = "phone", normalize the number
+        elif values.get("system") == "phone":
+            try:
+                # Attempt to parse with country code
+                if values["value"].startswith("+"):
```

Comment:
This is going to raise a server error if the user doesn't send the data in the correct format.  For example start up the local_server.sh and run this.

```sh
curl -X POST -H "Content-Type: application/json" -d '{"record": {"telecom": [{"system": "phone","val": " 555-123-4567"}]}}' http://localhost:8000/link
```

Reply:
Ah, I switched the mode to "after", which should take care of this issue the model validator will now run after the whole model has been validated.